airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_table.py

```diff
@@ -171,6 +171,176 @@ def test_given_three_pages_read_then_return_records(self, uuid_mock, mock_auth,
+        _given_get_timezone(http_mocker)
+        _given_read_schema(http_mocker)
+        _given_table_catalog(http_mocker)
+        _given_table_with_primary_keys(http_mocker)
+        http_mocker.post(
+            table_request().with_table(_TABLE).with_requestID(_REQUESTID).with_async().build(),
+            [
+            snowflake_response("async_response",FieldPath("statementStatusUrl"))
+            .with_handle(_HANDLE)
+            .with_status_code(error_code)
+            .build(),
+            snowflake_response("async_response",FieldPath("statementStatusUrl"))
+            .with_handle(_HANDLE)
+            .build()
+            ]
+        )
+        http_mocker.get(
+            table_request().with_handle(_HANDLE).build(is_get=True),
+            snowflake_response("response_get_table",JsonPath("$.'data'")).with_record(a_snowflake_response("response_get_table",JsonPath("$.'data'[*]"))).build()
+        )
+        output = _read(_config(), sync_mode=SyncMode.full_refresh)
+
+    
+
+    @parameterized.expand(
+            [
+                (429),
+                (202),
+                (random.randrange(500, 600)),
+                (random.randrange(500, 600)),
+            ],
+    )
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.SnowflakeStream.retry_factor", return_value=0, new_callable=mock.PropertyMock)
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.uuid.uuid4", return_value=_REQUESTID)
+    @mock.patch("source_snowflake.source.SnowflakeJwtAuthenticator")
+    @HttpMocker()
+    def test_retry_on_get_error(
+        self, 
+        error_code, 
+        property_mock, 
+        uuid_mock, 
+        mock_auth, 
+        http_mocker: 
+        HttpMocker) -> None:
+        _given_get_timezone(http_mocker)
+        _given_read_schema(http_mocker)
+        _given_table_catalog(http_mocker)
+        _given_table_with_primary_keys(http_mocker)
+
+        http_mocker.post(
+            table_request()
+            .with_table(_TABLE)
+            .with_requestID(_REQUESTID)
+            .with_async()
+            .build(),
+            snowflake_response("async_response",FieldPath("statementStatusUrl"))
+            .with_handle(_HANDLE)
+            .build()
+        )
+        http_mocker.get(
+            table_request()
+            .with_handle(_HANDLE)
+            .build(is_get=True),
+            [
+            snowflake_response("response_get_table",JsonPath("$.'data'"))
+            .with_record(a_snowflake_response("response_get_table",JsonPath("$.'data'[*]")))
+            .with_status_code(error_code).build(),
+            snowflake_response("response_get_table",JsonPath("$.'data'"))
+            .with_record(a_snowflake_response("response_get_table",JsonPath("$.'data'[*]"))).build(),
+            ]
+        )
+        output = _read(_config(), sync_mode=SyncMode.full_refresh)
```

Comment:
What is the assertion here? Should we validate that there are no errors and that we have a record?

Reply:
i thought validating that those code triger a retry was sufficiant.

Reply:
done